http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/internal/response/InputStreamAdaptingHttpStreamResponseHandler.java

```diff
@@ -121,16 +128,15 @@ private void onFailedResponseComplete(HttpStream stream, int errorCode) {
     }
 
     private void onSuccessfulResponseComplete(HttpStream stream) {
-        // always close the connection on a 5XX response code.
-        if (HttpStatusFamily.of(responseBuilder.statusCode()) == HttpStatusFamily.SERVER_ERROR) {
-            crtConn.shutdown();
-        }
-
         // For response without a payload, for example, S3 PutObjectResponse, we need to complete the future
         // in onResponseComplete callback since onResponseBody will never be invoked.
         requestCompletionFuture.complete(responseBuilder.build());
-        simplePublisher.complete();
-        crtConn.close();
-        stream.close();
+        simplePublisher.complete().whenComplete((result, failure) -> {
+            if (failure != null) {
+                failFutureAndReleaseConnection(stream, failure);
```

Comment:
Just confused: requestCompletionFuture.complete was invoked at line 133, and this line will call completeExceptionally again, [here](https://github.com/aws/aws-sdk-java-v2/pull/4825/files#diff-b2bbbb71a4e2ac299b2395fd7a94f7511e542dee488d6c2a5a505c13b46fbeb7R113). What behavior are we expecting?

Reply:
Good catch, will fix it.